agent/taskresource/volume/dockervolume.go

```diff
@@ -387,3 +398,67 @@ func (vol *VolumeResource) UnmarshalJSON(b []byte) error {
 	}
 	return nil
 }
+
+// updateAppliedStatusUnsafe updates the resource transitioning status
+func (vol *VolumeResource) updateAppliedStatusUnsafe(knownStatus resourcestatus.ResourceStatus) {
```

Comment:
seems like this isn't used anywhere. did you intend to use it in `UpdateAppliedStatus`?